catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.crashlytics.android.Crashlytics;
+import com.crashlytics.android.core.CrashlyticsCore;
+import com.google.gson.Gson;
+
+import org.catrobat.catroid.BuildConfig;
+import org.catrobat.catroid.ui.MainMenuActivity;
+import org.catrobat.catroid.ui.SettingsActivity;
+
+import io.fabric.sdk.android.Fabric;
+
+public final class CrashReporter {
+
+	private static final String TAG = CrashReporter.class.getSimpleName();
+
+	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
+	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+
+	static Context context;
+	private static SharedPreferences preferences;
+	private static boolean isCrashReportEnabled;
+	private static boolean isCrashlyticsInitialized;
+
+	private CrashReporter() {
+	}
+
+	public static boolean initialize(Context context) {
+
+		CrashReporter.context = context;
+		preferences = PreferenceManager.getDefaultSharedPreferences(context);
+		setIsCrashReportEnabled(BuildConfig.CRASHLYTICS_CRASH_REPORT_ENABLED);
+		isCrashlyticsInitialized = false;
```

Comment:
Please remove, `isCrashlyticsInitialized` is set later in this method to either `true` or `false`